docs/auth.md

```diff
@@ -0,0 +1,66 @@
+### KVS SDK Auth integration
+
+KVS SDK defines a generic Credential Provider object defined in https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/include/com/amazonaws/kinesis/video/common/Include.h#L452 which is used by the core of the SDK to get the credentials whenever they are needed. The abstract AwsCredentialProvider object contains a single API called GetCredentialsFunc which is called by the core SDK to fetch/refresh the credentials.
+
+The Credential Provider returns an AWS Credentials object defined in https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/include/com/amazonaws/kinesis/video/common/Include.h#L296 which will be used for the backend call authentication. It contains fields for Access Key ID, Secret Key and an optional Session Token. The credentials object also contains UINT64 expiration field which is the absolute time in the 100ns in the Producer device system clock when the credentials will expire. The core of the SDK will refresh the credentials whenever they are needed to be refreshed for the backend API calls by calling GetCredentialsFunc on the Credential Provider object.
+
+Kinesis Video Streams uses IAM policies/roles for access control as described in AWS documentation: https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/how-iam.html
+
+KVS can integrate with a variety of authentication providers via the Credential Provider object. Currently, the SDK defines the following Credential Providers. An application can instantiate the appropriate provider by calling create<Name>CredentialProvider API and submit it to the SDK. The provider can be freed on application termination after the core of the SDK has been freed by calling the corresponding free<Name>CredentialProvider API.
+
+#### Static credential provider
+
+Static credential provider is used whenever the application needs to integrate with a "persistent" Access Key ID/Secret Key pair. It doesn't have any expiration and will vend out the same credentials with every call.
+
+https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/include/com/amazonaws/kinesis/video/common/Include.h#L500
+
+#### IoT based credential provider
+
+Most of the devices will likely integrate with IoT certificates model during the provisioning to avoid "baking" the static credentials into the device firmare. More information on how to generate the IoT certificate can be found in AWS documentation: https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/how-iot.html
```

Comment:
Why do we prefer IoT credential provider over static? I think people will be curious about this.

Reply:
Added some info